### ISSUE TITLE
Image rotation based on exif

### ISSUE THREAD

**Issue by MichelFalke** [sulu/sulu#4088]
| Q | A
| --- | ---
| Bug? | no
| New Feature? | yes
| Sulu Version | <= 1.6.20, dev-develop
| Browser Version | All

#### Actual Behavior

Currently the ImagineImageConverter doesn't use the Autorotate filter which takes exif data into consideration. So images won't rotate based on the exif data.

#### Expected Behavior

Rotate images based on the orientation of the picture

#### Steps to Reproduce

Add an image which uses a different rotation then the "default top-left" (see the test image)

#### Possible Solutions

Hook into the ImagineImageConverter and always add the rotation filter.
[test-image.zip](https://github.com/sulu/sulu/files/2249652/test-image.zip)

**Comment by alexander-schranz** [comment]
seems like [autorotate](https://github.com/avalanche123/Imagine/blob/91ebbe7c9462a3df05bc5312c49c21aa34fa4756/lib/Imagine/Filter/Basic/Autorotate.php#L22) needs an additional PHP extension to read the exif data:

So think we will need to add the following:

```php
private function autorotate(ImageInterface $image)
{
     if (!ExifMetadataReader:isSupported) {
          return $image;
     }

     $autorotateFilter = new Autorotate();
     return $autorotateFilter->apply($image);
}
```

and call this function before the image is resized e.g. after [toRgb](https://github.com/sulu/sulu/blob/737820f260504f7a3a33517d47ebb44b95359b55/src/Sulu/Bundle/MediaBundle/Media/ImageConverter/ImagineImageConverter.php#L122) function.

**Comment by danrot** [comment]
@alexander-schranz If you say so :-) @MichelFalke Would you have time to provide a PR?

**Comment by MichelFalke** [comment]
@danrot I will

**Comment by alexander-schranz** [comment]
@MichelFalke you still want to contribute this. [Hacktoberfest](https://hacktoberfest.digitalocean.com) is the best time for it ;)

**Comment by alexander-schranz** [comment]
this issue is fixed in 1.6.23
